sources/core/Stride.Core/Utilities.cs

```diff
@@ -34,6 +34,50 @@ namespace Stride.Core;
 /// </summary>
 public static class Utilities
 {
+    // MUST BE A METHOD AND AGGRESSIVELY INLINED, OTHERWISE THE JIT WILL NOT ELIMINATE THE BRANCH
+    [MethodImpl(MethodImplOptions.AggressiveInlining)] 
+    private static bool IsUnalignedSafe() =>
+        RuntimeInformation.ProcessArchitecture is Architecture.X64 or Architecture.X86 or Architecture.Arm64;
+
+    /// <inheritdoc cref="AlignmentFallbackDoc"/>
+    /// <inheritdoc cref="Unsafe.CopyBlock(ref byte, ref readonly byte, uint)"/>
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    public static unsafe void CopyWithAlignmentFallback(ref byte destination, ref readonly byte source, uint byteCount)
+    {
+        if (IsUnalignedSafe())
+            fixed (void* src = &source, dst = &destination)
+                Buffer.MemoryCopy(src, dst, byteCount, byteCount);
```

Comment:
Why not use `Span<T>.CopyTo()` in these cases. [It seems](https://source.dot.net/System.Private.CoreLib/R/8b6f60fcf144a356.html) it does a `Buffer.Memmove` internally, the same as `Buffer.MemoryCopy` ([link](https://source.dot.net/System.Private.CoreLib/R/9335864fe015e780.html)).

Reply:
Also, for cases where we are sure the memory is aligned (like those where we control the type and ensure proper layouts), you also have the option of `Unsafe.CopyBlock`

Reply:
I chose Buffer.MemoryCopy because it's more straightforward than creating two spans and using `Span.CopyTo` between them. But as you said, they map to the same method, so if span does not provide anything else compared to buffer I think that this is kind of bikeshedd-y ?

Reply:
For these utility methods, it's enough to use `Buffer.MemoryCopy`, yes.
My point is at those call sites where we could be using a `Span`. Maybe we have more of those in the future, as more of the API is converted to use/accept spans.

Reply:
Yep